asllib/doc/ASLTypingReference.tex

```diff
@@ -267,65 +281,137 @@
 \chapter{Introduction}
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
-The purpose of this document is to describe, in a formal and authoritative way, which ASL programs are considered \emph{valid}.
-Validity is defined in terms of a \emph{type system}~\cite{TypeSystemsLucaCardelli}. That is, a set of \emph{typing rules}.
+The purpose of this document is to describe, in a formal and authoritative way, which ASL programs are considered \emph{well-typed}.
+Well-typedness is defined in terms of a \emph{type system}~\cite{TypeSystemsLucaCardelli}. That is, a set of \emph{typing rules}.
 
-\section{Abstract Syntax}
-An \emph{abstract syntax tree} (AST, for short) represents an ASL specification as a labelled structured tree.
+A \emph{type checker} is an implementation of the ASL type system, which first parses an ASL specification.
+If the parsing is successful, the result is an \emph{abstract syntax tree} (AST, for short),
+which represents an ASL specification as a labelled structured tree. Otherwise, the result is a syntax error
+and the type checker does not proceed.
 %
-The ASL type system is defined over the ASL abstract syntax, which is given in~\cite{ASLAbstractSyntaxReference}.
+The ASL syntax and AST is defined in~\cite{ASLAbstractSyntaxReference}.
 
-\section{Environments}
+When an ASL specification is successfully parsed, we refer to the resulting AST as the \emph{parsed AST}.
+The type checker apply the rules of the type system to the parsed AST. If it is successful, the specification
+is considered well-typed and the result is a pair consisting of
+a \emph{static environment} and a \emph{typed AST}. Otherwise, the type checker returns a type error.
+
+\paragraph{Related documents:}
+\begin{itemize}
+  \item The ASL Language Reference Manual~\cite{LRM} (LRM, for short) introduces the concrete syntax and intent
+  of all ASL language constructs.
+  Please note that the LRM will be retired in due course. For ease of reviewing, we currently indicate which statement
+  of the LRM the present rules correspond to.
+  \item The Abstract Syntax Reference~\cite{ASLAbstractSyntaxReference} defines the abstract syntax, parsed AST, and typed AST.
+  \item The ASL Semantics Reference~\cite{ASLSemanticsReference} defines all valid behaviors of a well-typed ASL specification.
+\end{itemize}
+
+\paragraph{Undestanding the Typing Formalization:}
+We assume basic familiarity with the ASL language constructs (see the Language Reference Manual~\cite{LRM}).
```

Comment:
please be mindful that the LRM will be retired so perhaps best to rephrase this already